zrml/prediction-markets/src/migrations.rs

```diff
@@ -347,6 +412,383 @@ mod tests {
+                    outcome: old_dispute.outcome.clone(),
+                    bond,
+                };
+                let res = new_disputes.try_push(new_dispute);
+                if res.is_err() {
+                    log::error!(
+                        "MoveDataToSimpleDisputes: Could not push dispute for market id {:?}",
+                        market_id
+                    );
+                }
+
+                // switch to new reserve identifier for simple disputes
+                let sd_pallet_id = zeitgeist_primitives::constants::SD_PALLET_ID;
+                let sd_reserve_id = sd_pallet_id.0;
+                let pm_pallet_id = zeitgeist_primitives::constants::PM_PALLET_ID;
+                let pm_reserve_id = pm_pallet_id.0;
+
+                // charge weight defensivly for unreserve_named
+                // https://github.com/open-web3-stack/open-runtime-module-library/blob/24f0a8b6e04e1078f70d0437fb816337cdf4f64c/tokens/src/lib.rs#L1516-L1547
+                total_weight = total_weight.saturating_add(T::DbWeight::get().reads_writes(4, 3));
+                let reserved_balance = <T as Config>::AssetManager::reserved_balance_named(
+                    &pm_reserve_id,
+                    Asset::Ztg,
+                    &old_dispute.by,
+                );
+                if reserved_balance < bond.saturated_into::<u128>().saturated_into() {
+                    log::error!(
+                        "MoveDataToSimpleDisputes: Could not unreserve {:?} for {:?} because \
+                         reserved balance is only {:?}. Market id: {:?}",
+                        bond,
+                        old_dispute.by,
+                        reserved_balance,
+                        market_id,
+                    );
+                }
+                <T as Config>::AssetManager::unreserve_named(
+                    &pm_reserve_id,
+                    Asset::Ztg,
+                    &old_dispute.by,
+                    bond.saturated_into::<u128>().saturated_into(),
+                );
+
+                // charge weight defensivly for reserve_named
+                // https://github.com/open-web3-stack/open-runtime-module-library/blob/24f0a8b6e04e1078f70d0437fb816337cdf4f64c/tokens/src/lib.rs#L1486-L1499
+                total_weight = total_weight.saturating_add(T::DbWeight::get().reads_writes(3, 3));
+                let res = <T as Config>::AssetManager::reserve_named(
+                    &sd_reserve_id,
+                    Asset::Ztg,
+                    &old_dispute.by,
+                    bond.saturated_into::<u128>().saturated_into(),
+                );
+                if res.is_err() {
+                    log::error!(
+                        "MoveDataToSimpleDisputes: Could not reserve bond for dispute caller {:?} \
+                         and market id {:?}",
+                        old_dispute.by,
+                        market_id
+                    );
+                }
```

Comment:
This part of the migration is not tested yet.

Reply:
https://github.com/zeitgeistpm/zeitgeist/pull/938/commits/84bb0214ec2348342ed0f9c02f9699bb1e390d43